transaction/src/validation/transaction_validator.rs

```diff
@@ -85,190 +161,110 @@ impl TransactionValidator {
     }
 
     pub fn validate_intent<I: IntentHashManager>(
+        &self,
         intent: &TransactionIntent,
         intent_hash_manager: &I,
-        config: &ValidationConfig,
-    ) -> Result<Vec<ExecutableInstruction>, TransactionValidationError> {
+    ) -> Result<Vec<Instruction>, TransactionValidationError> {
         // verify intent hash
         if !intent_hash_manager.allows(&intent.hash()) {
             return Err(TransactionValidationError::IntentHashRejected);
         }
 
         // verify intent header
-        Self::validate_header(&intent, config)
+        self.validate_header(&intent)
             .map_err(TransactionValidationError::HeaderValidationError)?;
 
-        let mut instructions = vec![];
-
         // semantic analysis
         let mut id_validator = IdValidator::new();
         for inst in &intent.manifest.instructions {
             match inst.clone() {
-                Instruction::TakeFromWorktop { resource_address } => {
+                Instruction::TakeFromWorktop { .. } => {
                     id_validator
                         .new_bucket()
                         .map_err(TransactionValidationError::IdValidationError)?;
-                    instructions.push(ExecutableInstruction::TakeFromWorktop { resource_address });
```

Comment:
I'm not sure I agree with this convergence of (Manifest)Instruction and (Executable)Instruction.

It removes the ability to have some instructions which can't be put into manifests / exposed to users => for someone writing a  Rust manifest, validation would be in code, not enforced at compile time.

Moreover, we will need (Manifest)Instructions to be versioned to a given Transaction version, whereas versioning of ExecutableInstructions will be tied to the Engine's execution format versioning. A given Engine version would have one type for ExecutableTransaction, but might need to support multiple Transaction formats. I really imagine we'd want to be able to version them separately.

Also, by changing this to be a clone at end, and not pushing instructions, we remove the ability to map Transaction Instructions to Executable Instructions.